### PR TITLE
Add support for using Docker for local development

### PR DESCRIPTION
**Pull request by patjouk**
This PR introduces support for using Docker for local development. How to use it is described in the [README](https://github.com/mozilla/foundation.mozilla.org/tree/docker-local-dev#how-to-setup-your-dev-environment-with-docker) and the [Docker for local dev](https://github.com/mozilla/foundation.mozilla.org/blob/docker-local-dev/docs/local_development_with_docker.md) documentation. Training for staff needs to be scheduled.

Moving to Docker will copy your existing `.env` to the `.docker.env` file that is in charge of managing env vars for Docker. You will get a new database populated with fake data: the setup task will invite you to create a new admin.

Run `invoke docker-setup` and give me feedback :)

# Changes impacting everyone
- Cypress unzipping was messing up output rendering with Docker, I moved the install as a separate npm command for now. It might change when this [PR gets merged.](https://github.com/cypress-io/cypress/pull/2706),
- Removed -f and -o flags from invoke commands: those were terrible, thanks past me! Use quotes instead ex: `inv manage "load_fake_data --delete"`,
- New invoke commands: if you don't want to use Docker, don't use the `docker-X` invoke tasks. All previous commands work as before.
- Catch(-)up command alias: I've added an alias for the catch(-)up commands because I kept forgetting if it was a dash or not, and I guess I'm not the only one. Both `invoke catchup` and `invoke catch-up` will work.

---

Closes MozillaFoundation/Docker-foundation-site#7
Closes MozillaFoundation/Docker-foundation-site#6
Closes MozillaFoundation/Docker-foundation-site#4

Related PRs/issues MozillaFoundation/Docker-foundation-site#2, MozillaFoundation/Docker-foundation-site#1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,7 +62,18 @@ When relevant, we encourage you to write tests. You can run the tests using the
 In addition to the code tests there are also visual regression tests, located in the `./cypress/integration` directory. You can run these tests locally by installing [cypress](https://www.cypress.io/) using `npm i cypress@3.0.3`, after which the command `npm run cypress` will run these tests locally. However, note that these tests are currently intended for screenshot comparisons across branches, and so will not yield any meaningful results when run for a single branch.
 
 
+## How to Setup your Dev Environment with Docker
+
+- Install [Docker Desktop](https://www.docker.com/products/docker-desktop) (macOS and Windows). For Linux users: install [Docker CE](https://docs.docker.com/install/#supported-platforms) and [Docker Compose](https://docs.docker.com/compose/install/),
```
Comment (youriwims) [inline README.md]: Maybe include the direct links to the install or to the comments in this [docker-login thread](https://github.com/docker/docker.github.io/issues/6910#issuecomment-398819417)

File: README.md [hunk 2]
```diff
@@ -57,7 +62,18 @@ When relevant, we encourage you to write tests. You can run the tests using the
...
+- [Check your install](https://docs.docker.com/get-started/#test-docker-version) by running `docker run hello-world`,
+- If relevant: delete your node_modules directory (`rm -rf node_modules`). It's not necessary, but it speeds up the install.
+- Run `invoke docker-setup` ([install invoke](http://www.pyinvoke.org/installing.html) if you don't have it yet). If you're running on Windows, you need to run `docker-compose --rm pipenv run python network-api/manage.py createsuperuser` when the setup is finished.
```
Comment (cadecairos) [inline README.md]: can't we alias that extra windows step with invoke? `invoke docker-setup-windows`?

File: README.md [hunk 3]
```diff
@@ -57,7 +62,18 @@ When relevant, we encourage you to write tests. You can run the tests using the
...
+
+This task is copying your `.env` to the new `.docker.env` that is in charge of managing your environment variables while running Docker. The installation will take a few minutes: you need to download images from the Docker Hub, install JS and Python dependencies, create fake data, migrate your database, etc.
+
+When it's done, run `docker-compose up`, wait until the static files to be built, and go to `0.0.0.0:8000`. You should have a local working version of the foundation site with fake data. When you want to stop, do `^C` to shut down your containers.
```
Comment (cadecairos) [inline README.md]: we could also mention the --detached flag to run it all as a background service. (`docker-compose logs --follow backend` to see the django logs)

**Comment by Pomax** [comment]
When running `invoke docker-setup` I eventually crash out with the error:
```
Creating foundationmozillaorg_watch-static-files_1 ... 
Creating foundationmozillaorg_postgres_1           ... error

Creating foundationmozillaorg_watch-static-files_1 ... done
r 0.0.0.0:5432 failed: port is already allocated

ERROR: for postgres  Cannot start service postgres: driver failed programming external connectivity on endpoint foundationmozillaorg_postgres_1 (04948c815f8024bd89af72600cbe40dd478d5223c3cae99a11bfb76dea540e0d): Bind for 0.0.0.0:5432 failed: port is already allocated
ERROR: Encountered errors while bringing up the project.
```
Which makes sense, I have postgres installed because I need it for other things as well - probably worth seeing if we can port-detect and use a different port is 5432 is taken?

File: README.md (hunk 2 above)
Comment (patjouk) [reply README.md]: Sadly no, for the same reason it was not possible to do it before.

**Comment by patjouk** [comment]
I updated docs and stop exposing postres port: that way, if postgres already run on the host, it won't fail anymore :)

File: docker-compose.yml [hunk 4]
```diff
@@ -9,6 +9,8 @@ services:
     environment:
       # Need to specify the SHELL env var for chokidar
       - SHELL=/bin/sh
+      # Force polling because inotify doesn't work on Docker Windows
+      - CHOKIDAR_USEPOLLING=1
```
Comment (cadecairos) [inline docker-compose.yml]: works great on Windows now!

**Comment by patjouk** [comment]
@cadecairos Ready for a final review before merging.

**Pull request merged**